pkg/bindings/kafka_sarama/message.go

```diff
@@ -0,0 +1,153 @@
+package kafka_sarama
+
+import (
+	"bytes"
+	"context"
+	"strings"
+
+	ce "github.com/cloudevents/sdk-go"
+	"github.com/cloudevents/sdk-go/pkg/binding"
+	"github.com/cloudevents/sdk-go/pkg/binding/format"
+	"github.com/cloudevents/sdk-go/pkg/binding/spec"
+
+	"github.com/Shopify/sarama"
+)
+
+const prefix = "ce_"
+
+var specs = spec.WithPrefix(prefix)
+
+const ContentType = "content-type"
+
+// Message holds a sarama ConsumerMessage.
+type Message struct {
```

Comment:
So... this is abstracting the actual `sarama.ConsumerMessage`, where the key/value pairs are basically the entire kafka ConsumerMsg?

Reply:
so, it's mixing Kafka content (`key`/`value`) with couldevent metadata?

Reply:
I've tried to abstract the `sarama.ConsumerMessage`, taking only the bits relevant for the cloudevents spec compliance. My hope is that, if we want to implement another kafka binding with another kafka library, this data structure could be potentially reused.
The `format` and `version` fields are here just to avoid walking again through the data structure after it was built the first time. They represent the [encoding of the `Message`](https://github.com/cloudevents/sdk-go/pull/314/files#diff-0c2a9f6c887a3e1374e37dfcfa4fba43R91) and they are reused later for [visiting the message](https://github.com/cloudevents/sdk-go/pull/314/files#diff-0c2a9f6c887a3e1374e37dfcfa4fba43R102)